### PR TITLE
Allow specifying sequences to clear as a list of ranges

### PR DESCRIPTION
**Pull request by romac**
See https://github.com/informalsystems/hermes/issues/3672 for the description of the feature and the syntax.

### CODE REVIEW

File: crates/relayer/src/link/cli.rs [hunk 1]
```diff
@@ -92,25 +93,23 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Link<ChainA, ChainB> {
...
-            )
-            .map_err(LinkError::supervisor)?
-        } else {
-            (
-                maybe_sequences,
-                self.a_to_b.src_chain().query_latest_height().unwrap(),
-            )
-        };
+        // Find the sequence numbers of unreceived packets
+        let (mut sequences, src_response_height) = unreceived_packets(
```
Comment (romac) [inline crates/relayer/src/link/cli.rs:97]: @ancazamfir I assumed that we want to allow half-open ranges (eg. `2000..` to mean all packets with sequence number 2000 or higher) or even just large ranges, we cannot enumerate all of them as there may be up to 2^64 of them. Therefore we have to query the unreceived packets/acks and then filter by the ranges we are given.

The downside is that we now do have to query for those even if we know exactly which packets we want to clear. The upside is that we won't attempt to query for packets which have already been relayed.

Does that trade-off work for you?

File: crates/relayer/src/link/cli.rs [hunk 2]
```diff
@@ -159,27 +158,26 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Link<ChainA, ChainB> {
...
-            )
-            .map_err(LinkError::supervisor)?
-        } else {
-            Some((
-                maybe_sequences,
-                self.a_to_b.src_chain().query_latest_height().unwrap(),
-            ))
-        }) else {
+        // Find the sequence numbers of unreceived acknowledgements
+        let Some((mut sequences, src_response_height)) = unreceived_acknowledgements(
```
Comment (romac) [inline crates/relayer/src/link/cli.rs:162]: Same as https://github.com/informalsystems/hermes/pull/3756/files#r1444722668

File: crates/relayer/src/link/cli.rs (hunk 1 above)
Comment (ancazamfir) [reply crates/relayer/src/link/cli.rs:97]: I think it should be fine. We can think of further optimizations in the future if needed.

**Review by ancazamfir — APPROVED** [review]
Very cool! Thanks @romac

**Pull request merged**